### DISCUSSION TITLE
Welcome to quadstore discussions!

### DISCUSSION THREAD

**Discussion by jacoscaz** [quadstorejs/quadstore#110] [General]
## 👋 Welcome!

  We’re using Discussions as a place to connect with developers using quadstore in their projects. We hope that you:

  * ask questions you’re wondering about;
  * share ideas;
  * engage with other community members, welcome others and keep an open-mind.

  To get started, comment below with an introduction of yourself and tell us about what you do with quadstore.

**Comment by simonegasperoni** [comment]
Good afternoon,
I m about to start a project that is supposed to heavily rely on the linked data. 
Since the users need to generate their own triples i was thinkg that a scalable approach would be to store locally those triples (kind of decentalized approach). 
The idea is providing the business logic by combining the user-triples and the server-side-triples (which would be possible by federating the endpoints 1:browser side and 2:server side)

Before starting digging into this ecosytem do you believe it possible to do?
is your sparql engine compatible with SPARQL 1.1?
basically the user should upload the rdf snippet (which could be very big) on the browser, then he executes federated queries to reach the  remote endpoint from the endpoint in the browser.
Is it possible?

many thanks,
Simone

**Reply by jacoscaz** [reply]
Hello @simonegasperoni ! Assuming I got it right, what you're proposing is definitely possible. Quadstore's SPARQL engine (the `quadstore-comunica` package) is built upon the [Comunica](https://comunica.dev) framework using a limited subset of components that do not allow for query federation. This was done intentionally to produce the smallest (in terms of bundle size) possible browser-side application bundle. 

However, Comunica definitely supports query federation and the latter is, in fact, one of its most distinguishing features. What I would suggest, therefore, is to start by looking at Comunica and perhaps speak with its lead maintainer @rubensworks. Thanks to the interoperability provided by the RDF/JS interfaces, you'd still be able to use `quadstore` as an isomorphic and persistent RDF store with any configuration of Comunica's components.

**Reply by simonegasperoni** [reply]
thanks for your message,
ok i ll start looking at comunica as first step
